### ISSUE TITLE
local docker fails: `Error: No such command '(lilac) hf-docker-start'`

### ISSUE THREAD

**Issue by devidw** [lilacai/lilac#722]
was running into some transformers decency issues when trying to run the project via the cli installed with pip

therefore was looking into getting it running inside a docker container

the dockerfile seems to depend on some local files that don't exist in the repo or i was not executing the commands that are expected to be run before building the docker image

so i made some slight adjustments to the dockerfile to have it build, but it seems like the main cli is missing an expected subcommand?

![Screenshot 2023-09-27 at 6 17 57 PM](https://github.com/lilacai/lilac/assets/68775926/be385de7-e1c0-4a3f-9eb1-24a68cf48b59)

```dockerfile
# NOTE: When we upgrade to 3.11 we can use a slimmer docker image which comes with gcc.
FROM python:3.9-bullseye

# Allow statements and log messages to immediately appear in the Knative logs
ENV PYTHONUNBUFFERED True

# See: https://huggingface.co/docs/hub/spaces-sdks-docker#permissions
RUN useradd -m -u 1000 user
USER user
ENV HOME=/home/user \
  PATH=/home/user/.local/bin:$PATH

# Set the working directory in the container.
WORKDIR $HOME/app

# Install the dependencies. This will look in ./dist for any wheels that match lilac. If they are
# not found, it will use the public pip package.

# Pip install lilac[all] and dependencies before trying to install the local image. This allows us
# to get cache hits on dependency installations when using a local wheel. When using the public pip
# package, the second call will be a no-op.
RUN python -m pip install lilac[all]

# Install from the local wheel inside ./dist. This will be a no-op if the wheel is not found.
#COPY --chown=user /dist ./dist/
#RUN python -m pip install --find-links=dist --upgrade lilac[all]
RUN python -m pip install lilac[all]

# Install the huggingface hub, used to download files.
RUN pip install huggingface_hub

# Copy the README so we can read the datasets from the HuggingFace config.
#COPY --chown=user README.md .
# Copy the license just in case.
#COPY --chown=user LICENSE .

COPY --chown=user docker_start.sh ./

# Make a local data directory for non-persistent storage demos.
RUN mkdir -p ./data
RUN chown -R user ./data

CMD ["bash", "docker_start.sh"]
```

**Comment by nsthorat** [comment]
Hey @devidw sorry about the confusion here, we actually *just* landed this PR right before you posted this. We hadn't yet released a new version.

I just released 0.2.1 which has that command now, so you should try again.

However, this docker image is very specific to huggingface spaces. If you wanna spin this up for yourself I would:

- Remove the stuff about `useradd`. This is HF-space specific:
```
# DELETE THIS!
# See: https://huggingface.co/docs/hub/spaces-sdks-docker#permissions
RUN useradd -m -u 1000 user
USER user
ENV HOME=/home/user \
  PATH=/home/user/.local/bin:$PATH
```
- For your own stuff, you can remove the double pip install. This is a perf thing so we get cached deps when we upload a local wheel for debugging.
- You can remove the `pip install huggingface_hub`
- In `docker_start.sh`, you can remove the line that says `lilac hf-docker-start`. This command syncs data dependencies in our docker image on HF before starting the server. Irrelevant for what you want.

### Other thoughts

You also could do this outside a docker image if you use a venv instead of a global pip install, which hopefully should fix your pip install issues. You might have something globally installed causing you issues.

Hope this helps, lmk if you have more problems.

**Comment by devidw** [comment]
Ohhh I see, thank you, @nsthorat 🙏🏻

I was able to solve the dep issue by clearing transformers cache and logging out of the huggingface-cli (was hitting some weird bad formatted request header with the hf_ token for some reason 😅)

Looking forward to exploring this project, using embeddings vs keywords working on dataset is true game-changer 

Thanks building this :)
